frontend/app/src/components/StatusWidget/StatusWidget.tsx

```diff
@@ -136,237 +106,244 @@ const RUNNING_MAN_DISPLAY_DELAY_TIME_MS = 500
-  }
+  useEffect(() => {
+    if (scriptRunState === ScriptRunState.RUNNING) {
+      setScriptChangedOnDisk(false)
+      setPromptHovered(false)
+    }
+  }, [scriptRunState])
 
-  private renderWidget(): ReactNode {
-    if (this.isConnected()) {
+  useEffect(() => {
+    if (isConnected()) {
       if (
-        this.props.scriptRunState === ScriptRunState.RUNNING ||
-        this.props.scriptRunState === ScriptRunState.RERUN_REQUESTED
+        scriptRunState === ScriptRunState.RUNNING ||
+        scriptRunState === ScriptRunState.RERUN_REQUESTED
       ) {
-        // Show scriptIsRunning when the script is actually running,
-        // but also when the user has just requested a re-run.
-        // In the latter case, the server should get around to actually
-        // re-running the script in a second or two, but we can appear
-        // more responsive by claiming it's started immemdiately.
-
-        this.showRunningManAfterInitialDelay(RUNNING_MAN_DISPLAY_DELAY_TIME_MS)
-        return this.renderScriptIsRunning()
-      }
-      if (this.state.scriptChangedOnDisk) {
-        return this.renderRerunScriptPrompt()
+        showRunningManAfterInitialDelay(RUNNING_MAN_DISPLAY_DELAY_TIME_MS)
       }
     }
-    if (this.props.scriptRunState === ScriptRunState.NOT_RUNNING) {
-      this.setState({ showRunningMan: false })
-    }
-
-    return this.renderConnectionStatus()
-  }
-
-  /** E.g. "Disconnected [Icon]" */
-  private renderConnectionStatus(): ReactNode {
-    const ui = StatusWidget.getConnectionStateUI(this.props.connectionState)
-    if (ui === undefined) {
-      return null
+    if (scriptRunState === ScriptRunState.NOT_RUNNING) {
+      setShowRunningMan(false)
     }
-
-    return (
-      <Tooltip content={ui.tooltip} placement={Placement.BOTTOM}>
-        <StyledConnectionStatus
-          className="stConnectionStatus"
-          data-testid="stConnectionStatus"
-        >
-          <Icon size="sm" content={ui.icon} />
-          <StyledConnectionStatusLabel
-            isMinimized={this.state.statusMinimized}
-          >
-            {ui.label}
-          </StyledConnectionStatusLabel>
-        </StyledConnectionStatus>
-      </Tooltip>
-    )
-  }
-
-  private static isNewYears(): boolean {
-    // Test if current date between 12/31 & 1/06
-    const currentDate = new Date()
-    const month = currentDate.getMonth()
-    const date = currentDate.getDate()
-
-    // Check if Dec 31st
-    if (month === 11 && date === 31) return true
-    // Check if Jan 1st through 6th
-    if (month === 0 && date <= 6) return true
-
-    return false
-  }
-
-  /** "Running... [Stop]" */
-  private renderScriptIsRunning(): ReactNode {
-    const minimized = this.state.statusMinimized
-    const stopRequested =
-      this.props.scriptRunState === ScriptRunState.STOP_REQUESTED
-    const stopButton = StatusWidget.promptButton(
+  }, [
+    scriptRunState,
+    connectionState,
+    showRunningManAfterInitialDelay,
+    isConnected,
```

Comment:
**suggestion:** As written right now, is only checking if the `isConnected` callback itself changes, but not if the returned value is changing, which I have a hunch is unexpected. If this is updated to utilize the suggestion for `isConnected` to be a `useMemo`, then this will be fixed.